app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,235 @@
+                          onChanged: (newDuration) {
+                            setState(() => _duration = newDuration);
+                          },
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 6),
+                          child: ElevatedButton(
+                            onPressed: () => submitForm(Navigator.of(context)),
+                            child: const Text('Share'),
+                          ),
+                        ),
+                        error != null
+                            ? Text(
+                                error!,
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor),
+                              )
+                            : Container(),
+                        success != null
+                            ? Text(
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends State<ShareSettings> {
+  final api = obtainApiClient();
+  late Future _shareHolders;
+
+  @override
+  void initState() {
+    super.initState();
+    _shareHolders = api?.getCurrentShares(widget.bike.id);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: _shareHolders,
+      builder: (BuildContext context, AsyncSnapshot snapshot) =>
+          StatefulBuilder(
+        builder: (context, setState) {
+          if (snapshot.connectionState != ConnectionState.done) {
+            return Center(
+                child: Column(children: const [
+              CircularProgressIndicator(),
+              Text("Loading shares"),
+            ]));
+          }
+
+          if (snapshot.hasData) {
+            if (snapshot.data.length == 0) {
+              return const Text('No share holders');
+            } else {
+              return ListView.builder(
+                itemCount: snapshot.data.length,
+                itemBuilder: (context, index) {
+                  return ListTile(
+                    title: Text(snapshot.data[index]["email"]),
+                    subtitle: Text(snapshot.data[index]["duration"] == null
+                        ? "Forever"
+                        : (snapshot.data[index]["duration"] ~/ 86400) > 1
+                            ? "${snapshot.data[index]["duration"] / 86400} days"
+                            : (snapshot.data[index]["duration"] ~/ 3600) > 1
+                                ? "${snapshot.data[index]["duration"] / 3600} hours"
+                                : "${snapshot.data[index]["duration"] / 60} minutes"),
```

Comment:
We should make a variable of `snapshot.data[index]` and `snapshot.data[index]["duration"]` so we don't have so much duplicated code

Reply:
> We should make a variable of `snapshot.data[index]` and `snapshot.data[index]["duration"]` so we don't have so much duplicated code

How do you want to create a variable of it?

Reply:
Were i've placed the line
![image](https://user-images.githubusercontent.com/15320763/187037811-2fba8043-49ef-4687-adb9-56db9d11ce63.png)

something like:
```dart
final duration = snapshot.data[index]["duration"];
```

Reply:
> Were i've placed the line ![image](https://user-images.githubusercontent.com/15320763/187037811-2fba8043-49ef-4687-adb9-56db9d11ce63.png)
> 
> something like:
> 
> ```dart
> final duration = snapshot.data[index]["duration"];
> ```

Done, thanks.

My tricky way of checking if it is longer then a day or hour etc, is this fine or do you know a better way?

Reply:
That's oke